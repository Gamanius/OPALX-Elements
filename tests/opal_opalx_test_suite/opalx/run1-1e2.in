OPTION, PSDUMPFREQ = 1;   // 6d data written every 300000 time steps (h5).
OPTION, STATDUMPFREQ = 1;  // Beam Stats written every 10 time steps (stat).
OPTION, BOUNDPDESTROYFQ=10; // Delete lost particles, if any
OPTION, AUTOPHASE=4;        // Autophase is on, and phase of max energy
                            // gain will be found automatically for cavities
Option, VERSION=10900;

Title, string="Solenoid-1 test";

//----------------------------------------------------------------------------
//Global Parameters

REAL rf_freq             = 1.3e3;     //RF frequency. (Hz)
REAL n_particles         = 1e2;      //Number of particles in simulation.
REAL beam_bunch_charge   = 1e-9;      //Charge of bunch. (C)

//Initial Momentum Calculation
REAL Edes    = 1e-3;  // initial energy in GeV
REAL gamma   = (Edes+EMASS)/EMASS;
REAL beta    = sqrt(1-(1/gamma^2));
REAL P0      = gamma*beta*EMASS;    //inital z momentum

//Printing initial energy and momentum to terminal output.
value , {Edes, P0, OPALVERSION};

//----------------------------------------------------------------------------
// Solenoids
//
// L:           Physcial element length (m)
// ELEMEDGE:    Physcial start of element (m)
// KS:          Solenoid strength (Rad/m)
// FMAPFM:      Field file (string)

// Note: OPAL scales the field file based on the max magnetic
// field value in the file, not Bz on axis. The max field
// value is normalized to 1 [T], and scaled with KS.
// i.e. The max value in the BF_559 file = 0.162544398 [T].
// Therefore, setting KS = 0.162544398 runs the magnet at max current.

REAL KSBF = 0.162544398;
if (OPALVERSION>10500)
   KSBF = KSBF/1.3528;

BF: Solenoid, L = 0.5, ELEMEDGE=0.0, KS = KSBF, FMAPFN = "../BF_550.T7";

SOLine: Line = (BF);

Dist: DISTRIBUTION, TYPE = GAUSS,
        SIGMAX = 0.001,
        SIGMAY = 0.001,
	    SIGMAZ = 0.001;

//----------------------------------------------------------------------------
// Define Field solvers
// The mesh sizes should be a factor of 2
// for most efficient space charge calculation.

FS1: Fieldsolver, Type=NONE,
            NX = 16, NY = 16, NZ = 16, // SC grid size is 32^3
            PARFFTX = false,
            PARFFTY = false,
            PARFFTZ = true,  // parallel in the z direction only
            BCFFTX = open,
            BCFFTY = open,
            BCFFTZ = open,
            BBOXINCR = 1,
            GREENSF = INTEGRATED;

//----------------------------------------------------------------------------
// Electron Beam Definition

BEAM1:  BEAM, PARTICLE = ELECTRON, pc = P0, NPART = n_particles,
        BFREQ = rf_freq, BCURRENT = beam_bunch_charge * rf_freq * 1e6, CHARGE = -1;

//----------------------------------------------------------------------------
// Simulate the beamline using TRACK and RUN.
// Note, different time steps are set based on the z location in the beam line.
// In the case below, 1.0e-13 is used for 0.0 to 0.4 m,
// and 3.0e-12 is used from 0.4 to 5 m.

TRACK, LINE = SOLine, BEAM = BEAM1, MAXSTEPS = 33, DT = {1.0e-10};
RUN, METHOD = "PARALLEL", BEAM = BEAM1,
    FIELDSOLVER = FS1, DISTRIBUTION = Dist;
ENDTRACK;
Quit;
    