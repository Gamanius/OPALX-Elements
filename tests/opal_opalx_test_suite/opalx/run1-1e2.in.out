Warning: Option 'run1-1e2.in' is not parsed by Ippl.
OPAL-X> configure done argc= 4
OPAL-X>               ____  _____        _       __   __
OPAL-X> 
OPAL-X> This is OPAL-X (Object Oriented Parallel Accelerator Library) Version 2024.1.00
OPAL-X>             git rev. ac1b7f5951630e232502575123a79f799192cf6e
OPAL-X> 
OPAL-X> 
OPAL-X>                      (c) PSI, http://amas.web.psi.ch
OPAL-X> 
OPAL-X> 
OPAL-X> This is the performance portable version of OPAL
OPAL-X> 
OPAL-X> Please send cookies, goodies or other motivations (wine and beer ... ) 
OPAL-X> to the OPAL developers opal@lists.psi.ch
OPAL-X> 
OPAL-X> Time: 14:04:57 date: 20/05/2025
OPAL-X> 
OPAL-X> * Host:   OpenMP
OPAL-X> * Device: OpenMP
OPAL-X> 
OPAL-X> * Reading input stream 'run1-1e2.in'
OPAL-X> 
OPAL-X> value: {EDES,P0,OPALVERSION} = {0.001, 0.00142197, 2.02401e+07}
OPAL-X> 
OPAL-X> * Number of distributions  1
OPAL-X> 
OPAL-X> * ************* D I S T R I B U T I O N ********************************************
OPAL-X> * 
OPAL-X> * Distribution type: GAUSS
OPAL-X> * 
OPAL-X> * SIGMAX     = 0.001 [m]
OPAL-X> * SIGMAY     = 0.001 [m]
OPAL-X> * SIGMAZ     = 0.001 [m]
OPAL-X> * SIGMAPX    = 0 [Beta Gamma]
OPAL-X> * SIGMAPY    = 0 [Beta Gamma]
OPAL-X> * SIGMAPZ    = 0 [Beta Gamma]
OPAL-X> * 
OPAL-X> * Distribution is injected.
OPAL-X> * 
OPAL-X> * **********************************************************************************
OPAL-X> 
OPAL-X> * ************* F I E L D S O L V E R ********************************************** 
OPAL-X> * FIELDSOLVER  FS1
OPAL-X> * TYPE         NONE
OPAL-X> * RANKS       1
OPAL-X> * NX           16
OPAL-X> * NY           16
OPAL-X> * NZ           16
OPAL-X> * BBOXINCR     1
OPAL-X> * GREENSF      INTEGRATED
OPAL-X> * XDIM         serial  
OPAL-X> * YDIM         serial  
OPAL-X> * ZDIM      parallel  
OPAL-X> * ********************************************************************************** 
OPAL-X> 
OPAL-X> * ************* B E A M ************************************************************ 
OPAL-X> * BEAM        BEAM1
OPAL-X> * PARTICLE    ELECTRON
OPAL-X> * REST MASS   0.000510999 [GeV]
OPAL-X> * CHARGE      -e * 1 
OPAL-X> * MOMENTUM    1.42197e+06 [eV/c]
OPAL-X> * MOMENTUM    0.00142197 [GeV/c]
OPAL-X> * CURRENT     1.3 [A]
OPAL-X> * FREQUENCY   1300 [MHz]
OPAL-X> * NPART       100
OPAL-X> * ********************************************************************************** 
OPAL-X> 
OPAL-X> PartBunch Constructor
OPAL-X> * Solver and Load Balancer set
OPAL-X> number_of_processors 88
OPAL-X> MPI_Comm_size 1
OPAL-X> * About to create particles ...
OPAL-X> * Particle creation done
OPAL-X> 
OPAL-X> * ************** B U N C H ********************************************************* 
OPAL-X> * PARTICLES       = 100
OPAL-X> * CHARGE          = -1.000000e-09 (Cb) 
OPAL-X> * INTEGRATOR      = LF2
OPAL-X> * MIN R (origin)  = (     -2.27991 ,     -2.70818 ,     -2.59444 ) [mm]
OPAL-X> * MAX R (max ext) = (      1.96084 ,      2.78154 ,      2.38662 ) [mm]
OPAL-X> * RMS R           = (      0.83676 ,      1.15826 ,      0.94411 ) [mm]
OPAL-X> * RMS P           = ( 0.000000e+00 , 0.000000e+00 , 2.664535e-15 ) [beta gamma]
OPAL-X> * Mean R: ( -2.276825e-20 , -9.974660e-20 , 3.794708e-20 ) [m]
OPAL-X> * Mean P: ( 0.000000e+00 , 0.000000e+00 , 2.782725e+00 ) [beta gamma]
OPAL-X> * MESH SPACING    = (    267.69707 ,    346.53869 ,    314.42960 ) [um]
OPAL-X> * COMPDOM INCR    = 1.000000e+00 (%) 
OPAL-X> * FIELD LAYOUT    = Domain = {[0:15:1],[0:15:1],[0:15:1]}
OPAL-X> Total number of boxes = 1
OPAL-X>     Box 0 {[0:15:1],[0:15:1],[0:15:1]}
OPAL-X> 
OPAL-X> * Centroid : 
OPAL-X> * -2.276825e-18 0.000000e+00 -9.974660e-18 0.000000e+00 3.794708e-18 2.782725e+02 
OPAL-X> * Cov Matrix : 
OPAL-X> * 7.001660e-07 0.000000e+00 -5.820260e-08 0.000000e+00 -5.831774e-08 -3.562970e-35 
OPAL-X> * 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 
OPAL-X> * -5.820260e-08 0.000000e+00 1.341572e-06 0.000000e+00 -7.047820e-08 6.548162e-35 
OPAL-X> * 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 0.000000e+00 
OPAL-X> * -5.831774e-08 0.000000e+00 -7.047820e-08 0.000000e+00 8.913350e-07 1.078521e-34 
OPAL-X> * -3.562970e-35 0.000000e+00 6.548162e-35 0.000000e+00 1.078521e-34 7.099748e-30 
OPAL-X> * * ********************************************************************************** 
OPAL-X> * Creating new file for data: 'run1-1e2.stat'
OPAL-X> * Creating new file for data: 'run1-1e2.lbal'
OPAL-X> * BF gone live
OPAL-X> ParallelTrack: momentum=  2.78273
OPAL-X> itsBunch_m->RefPartR_m= ( -2.27682e-20 , -9.97466e-20 , 3.79471e-20 )
OPAL-X> itsBunch_m->RefPartP_m= ( 0 , 0 , 2.78273 )
OPAL-X> OrbitThreader dt_m= 1e-10
OPAL-X> OrbitThreader maxDistance= 1.79769e+308
OPAL-X> OrbitThreader #elements  = 1
OPAL-X> OrbitThreader maxDistance= 1.78996
OPAL-X> OrbitThreader #elements  = 0
OPAL-X> 
OPAL-X> * Size of map 2 sections 
OPAL-X> * Key: (      0.000000 -       0.507831) number of overlapping elements 1
OPAL-X> *                                        BF
OPAL-X> * Key: (      0.507831 - 1000006.000000) number of overlapping elements 0
OPAL-X> 
OPAL-X> * Creating new file for data: 'data/run1-1e2_ElementPositions.sdds'
OPAL-X> * Track start at: 14:04:57, t= 0.000 [fs]; zstart at: 0.000 [um]
OPAL-X> * Executing ParallelTracker
OPAL-X> * Initial dt = 100.000 [ps]
OPAL-X> * Max integration steps = 33, next step = 0
OPAL-X> 
OPAL-X>                dt [ns]           zStop [m]        num Steps [1]
OPAL-X> ->         1.000000e-10        1.000000e+06                  33
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      0 at 28.213 [mm], t= 100.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      1 at 56.426 [mm], t= 200.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      2 at 84.639 [mm], t= 300.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      3 at 112.851 [mm], t= 400.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      4 at 141.064 [mm], t= 500.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      5 at 169.277 [mm], t= 600.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      6 at 197.490 [mm], t= 700.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      7 at 225.703 [mm], t= 800.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      8 at 253.916 [mm], t= 900.000 [ps], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step      9 at 282.128 [mm], t= 1.000 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     10 at 310.341 [mm], t= 1.100 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     11 at 338.554 [mm], t= 1.200 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     12 at 366.767 [mm], t= 1.300 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     13 at 394.980 [mm], t= 1.400 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     14 at 423.193 [mm], t= 1.500 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     15 at 451.406 [mm], t= 1.600 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     16 at 479.618 [mm], t= 1.700 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     17 at 507.831 [mm], t= 1.800 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     18 at 536.044 [mm], t= 1.900 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     19 at 564.257 [mm], t= 2.000 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:57 Step     20 at 592.470 [mm], t= 2.100 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     21 at 620.683 [mm], t= 2.200 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     22 at 648.895 [mm], t= 2.300 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     23 at 677.108 [mm], t= 2.400 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     24 at 705.321 [mm], t= 2.500 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     25 at 733.534 [mm], t= 2.600 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     26 at 761.747 [mm], t= 2.700 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     27 at 789.960 [mm], t= 2.800 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     28 at 818.173 [mm], t= 2.900 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     29 at 846.385 [mm], t= 3.000 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     30 at 874.598 [mm], t= 3.100 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     31 at 902.811 [mm], t= 3.200 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> no solver avaidable 
OPAL-X> * 14:04:58 Step     32 at 931.024 [mm], t= 3.300 [ns], E=1.000 [MeV]
OPAL-X> * Wrote beam statistics.
OPAL-X> * Dump phase space of last step
OPAL-X> * BF gone off
OPAL-X> 
OPAL-X> * Done executing ParallelTracker at 14:04:58
OPAL-X> 
OPAL-X> * Finished time step: 0 time: 0.000000e+00
OPAL-X> * End of input stream 'run1-1e2.in'
Timings> ---------------------------------------------
Timings>      Timing results for 1 nodes:
Timings> ---------------------------------------------
Timings> mainTimer........... Wall tot =   0.592291
Timings> 
Timings> gatherInfoPartBunch. Wall max = 0.00213246
Timings>                      Wall avg = 0.00213246
Timings>                      Wall min = 0.00213246
Timings> 
Timings> setSolver........... Wall max = 0.000492717
Timings>                      Wall avg = 0.000492717
Timings>                      Wall min = 0.000492717
Timings> 
Timings> prerun.............. Wall max = 0.000134462
Timings>                      Wall avg = 0.000134462
Timings>                      Wall min = 0.000134462
Timings> 
Timings> samplingTime........ Wall max = 0.000726463
Timings>                      Wall avg = 0.000726463
Timings>                      Wall min = 0.000726463
Timings> 
Timings> SamplingTimer....... Wall max = 0.000674872
Timings>                      Wall avg = 0.000674872
Timings>                      Wall min = 0.000674872
Timings> 
Timings> GenParticles........ Wall max = 0.000679203
Timings>                      Wall avg = 0.000679203
Timings>                      Wall min = 0.000679203
Timings> 
Timings> particleBC.......... Wall max = 3.3229e-05
Timings>                      Wall avg = 3.3229e-05
Timings>                      Wall min = 3.3229e-05
Timings> 
Timings> updateParticle...... Wall max =          0
Timings>                      Wall avg =          0
Timings>                      Wall min =          0
Timings> 
Timings> Write Stat.......... Wall max = 0.00786822
Timings>                      Wall avg = 0.00786822
Timings>                      Wall min = 0.00786822
Timings> 
Timings> Write H5-File....... Wall max =   0.461027
Timings>                      Wall avg =   0.461027
Timings>                      Wall min =   0.461027
Timings> 
Timings> TIntegration1....... Wall max = 0.000120799
Timings>                      Wall avg = 0.000120799
Timings>                      Wall min = 0.000120799
Timings> 
Timings> TIntegration2....... Wall max = 0.000422337
Timings>                      Wall avg = 0.000422337
Timings>                      Wall min = 0.000422337
Timings> 
Timings> External field eval. Wall max = 0.00766134
Timings>                      Wall avg = 0.00766134
Timings>                      Wall min = 0.00766134
Timings> 
Timings> Binaryrepart........ Wall max =          0
Timings>                      Wall avg =          0
Timings>                      Wall min =          0
Timings> 
Timings> OrbThreader......... Wall max =  0.0253382
Timings>                      Wall avg =  0.0253382
Timings>                      Wall min =  0.0253382
Timings> 
Timings> ---------------------------------------------
